app/assets/javascripts/discourse/app/services/body-classes.js

```diff
@@ -8,30 +8,29 @@ export default class BodyClassesService extends Service {
 
   registerClasses(helper, classes) {
     if (this.#helpers.has(helper)) {
-      const classesToRemove = this.#helpers
-        .get(helper)
-        .filter((c) => !classes.includes(c));
+      const classesToRemove = this.#helpers.get(helper);
+
+      this.#helpers.set(helper, classes);
       this.removeClasses(classesToRemove);
     } else {
+      this.#helpers.set(helper, classes);
       registerDestructor(helper, () => {
         const removedClasses = this.#helpers.get(helper);
         this.#helpers.delete(helper);
-
         this.removeClasses(removedClasses);
       });
     }
 
-    this.#helpers.set(helper, classes);
     for (const bodyClass of classes) {
       document.body.classList.add(bodyClass);
     }
   }
 
   removeClasses(classes) {
```

Comment:
Maybe rename to something `maybeRemoveClasses()`? It confused me a bit because I saw calls to `.removeClasses` and thought "wait, some of these shouldn't be removed 👀"

Reply:
Haha, I was struggling with naming this function (every single variable in file 😂)
"maybeRemoveClasses", "classesToBeRemovedUnlessStillUsed", "classesButNotThoseFromTheArgs" 😄